website/src/mode.tsx

```diff
@@ -55,85 +53,69 @@ const cherokeeRepresentationMapping = {
   },
 }
 
-export const cherokeeRepresentationDetails = (
-  system: Dailp.CherokeeOrthography
-) => cherokeeRepresentationMapping[system]
-
-export const ExperiencePicker = (p: {
-  onSelect: (mode: ViewMode) => void
-  id?: string
-}) => {
-  const { viewMode: value, setViewMode: setValue } = usePreferences()
-
-  // Save the selected view mode throughout the session.
-  useEffect(() => {
-    p.onSelect(value)
-  }, [value])
-  return (
-    <Select
-      name="experience-picker"
-      id={p.id}
-      value={value}
-      onChange={(e) => setValue(Number.parseInt(e.target.value))}
-    >
-      {Object.entries(levelNameMapping).map(([viewMode, details]) => (
-        <option value={viewMode} key={viewMode}>
-          {details.label}
-        </option>
-      ))}
-    </Select>
-  )
-}
+export const SelectLevelOfDetail = (p: {
+  id: string
+  value: LevelOfDetail
+  onSelect: (mode: LevelOfDetail) => void
+}) => (
+  <Select
+    id={p.id}
+    name="experience-picker"
+    value={p.value}
+    onChange={(e) => p.onSelect(Number.parseInt(e.target.value))}
+  >
+    {Object.entries(levelNameMapping).map(([levelOfDetail, details]) => (
+      <option
+        value={levelOfDetail}
+        key={levelOfDetail}
+        aria-description={details.details}
+      >
+        {details.label}
+      </option>
+    ))}
+  </Select>
+)
 
-export const PhoneticsPicker = (p: {
+export const SelectCherokeeRepresentation = (p: {
+  id: string
+  value: Dailp.CherokeeOrthography
   onSelect: (phonetics: Dailp.CherokeeOrthography) => void
-}) => {
-  const { cherokeeRepresentation: value, setCherokeeRepresentation: setValue } =
-    usePreferences()
-
-  // Save the selected representation throughout the session.
-  useEffect(() => {
-    p.onSelect(value)
-  }, [value])
-  return (
-    <Select
-      name="phonetics-picker"
-      value={value}
-      onChange={(e) => setValue(e.target.value as Dailp.CherokeeOrthography)}
-      aria-label="Romanization"
-    >
-      {Object.entries(cherokeeRepresentationMapping).map(
-        ([system, details]) => (
-          <option value={system} key={system}>
-            {details.label}
-          </option>
-        )
-      )}
-    </Select>
-  )
-}
+}) => (
+  <Select
+    id={p.id}
+    name="cherokee-representation"
+    value={p.value}
+    onChange={(e) => p.onSelect(e.target.value as Dailp.CherokeeOrthography)}
```

Comment:
I'm not sure if this casting here is needed since the given prop value passed into this component's `value` is already a `Dailp.CherokeeOrthography`.

Reply:
Currently the `e.target.value` comes straight from the browser API, where a select value is always a string type. It doesn't adapt its type based on the value type passed in.